sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -626,6 +627,40 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     client.getPartition(db, table, spec)
   }
 
+  override def listPartitionsByFilter(
```

Comment:
> Why can't we just call client.getPartitionsByFilter in this method and put all other logic at caller side?

There are two reasons:

1. This method is intended to return exactly the partitions matching the given filter.
2. This method is called from multiple places. I feel this is an acceptable place to centralize this common logic.